rtt/transports/corba/RemotePorts.cpp

```diff
@@ -194,6 +189,20 @@ RTT::base::ChannelElementBase::shared_ptr RemoteInputPort::buildRemoteChannelOut
     return corba_ceb;
 }
 
+bool RemoteInputPort::disconnect(PortInterface* port)
+{
+    //just check the type and not relay on
+    RemoteOutputPort *oPort = dynamic_cast<RemoteOutputPort* >(port);
+    if(oPort == NULL){
+        Logger::In in("RemoteInputPort::disconnect(PortInterface& port)");
+        log(Error) << "Port: " << this->getName() << " could not be disconnected from: " << oPort->getName()
+                << " because it is not of type: RemoteOutputPort" << endlog();
+        return false;
+    }
+
+    return dataflow->removeConnection(oPort->getName().c_str(), this->getDataFlowInterface(), this->getName().c_str());
```

Comment:
> more natural to do that from the "view" of the output port

You really are not explaining anything to me. What's natural is currently irrelevant, I don't even understand why it should *work*.

Let's settle this one easy way: write unit tests, we'll see what works and what does not.